### PR TITLE
Improve usability of MultipleContentTypeField

### PR DESCRIPTION
**Pull request by glennmatthews**
### Fixes: #98 

Just some basic code cleanup. Makes it a lot simpler to use a `MultipleContentTypeField` or `CSVMultipleContentTypeField`.

### CODE REVIEW

File: nautobot/extras/forms.py [hunk 1]
```diff
@@ -807,12 +776,7 @@ class StatusFilterForm(BootstrapMixin, CustomFieldFilterForm):
 
     model = Status
     q = forms.CharField(required=False, label="Search")
-    content_types = MultipleContentTypeField(
-        queryset=ContentType.objects.filter(FeatureQuery("statuses").get_query()).order_by("app_label", "model"),
-        required=False,
-        label="Content type(s)",
-        widget=StaticSelect2Multiple(),
-    )
+    content_types = MultipleContentTypeField(feature="statuses", required=False, label="Content type(s)")
```
Comment (jathanism) [inline nautobot/extras/forms.py:779]: Something isn't quite right w/ this change. I'm trying to figure out why.

Add_a_new_status_-_Nautobot

File: nautobot/extras/forms.py [hunk 2]
```diff
@@ -821,11 +785,7 @@ class StatusBulkEditForm(BootstrapMixin, CustomFieldBulkEditForm):
 
     pk = forms.ModelMultipleChoiceField(queryset=Status.objects.all(), widget=forms.MultipleHiddenInput)
     color = forms.CharField(max_length=6, required=False, widget=ColorSelect())
-    content_types = forms.ModelMultipleChoiceField(
-        queryset=ContentType.objects.filter(FeatureQuery("statuses").get_query()).order_by("app_label", "model"),
-        label="Content type(s)",
-        required=False,
-    )
+    content_types = MultipleContentTypeField(feature="statuses", required=False, label="Content type(s)")
```
Comment (jathanism) [inline nautobot/extras/forms.py:788]: Bulk edit is experiencing the same issue:

Editing_2_Statuses_-_Nautobot

**Review by jathanism — CHANGES_REQUESTED** [review]
I'm not sure what the changes are yet. I just know that something is missing. It appears that the string value vs the primary key is being populated in the form field?

**Comment by jathanism** [comment]
Okay in looking at it, it appears to only be on the `StatusForm` and `StatusBulkEditForm` and it's because these were moved from `ModelMultipleChoiceField` to `MultipleContentTypeField`, and because `MultipleContentTypeField._generate_choices_from_queryset()` is used to explicitly set the `choices` mapping to the string representation vs. pk.

**Comment by jathanism** [comment]
Here's a fix:

`nautobot/utilities/forms/fields.py`
```diff
diff --git a/nautobot/utilities/forms/fields.py b/nautobot/utilities/forms/fields.py
index 51ffc1b0..b6387af0 100644
--- a/nautobot/utilities/forms/fields.py
+++ b/nautobot/utilities/forms/fields.py
@@ -186,11 +186,13 @@ class MultipleContentTypeField(forms.ModelMultipleChoiceField):
             )
         if "widget" not in kwargs:
             kwargs["widget"] = widgets.StaticSelect2Multiple()
+        self.choices_as_strings = kwargs.pop("choices_as_strings", True)

         super().__init__(*args, **kwargs)

         # Generate choices from queryset each time form is initialized.
-        self.choices = self._generate_choices_from_queryset
+        if self.choices_as_strings:
+            self.choices = self._generate_choices_from_queryset

     def _generate_choices_from_queryset(self):
         """Overload choices to return "<app>.<model>" for CSV import help text."""
(END)
```

`nautobot/extras/forms.py`
```diff
diff --git a/nautobot/extras/forms.py b/nautobot/extras/forms.py
index 320b9a8e..44a78195 100644
--- a/nautobot/extras/forms.py
+++ b/nautobot/extras/forms.py
@@ -700,7 +700,9 @@ class CustomLinkFilterForm(BootstrapMixin, forms.Form):


 class WebhookForm(BootstrapMixin, forms.ModelForm):
-    content_types = MultipleContentTypeField(feature="webhooks", required=False, label="Content Type(s)")
+    content_types = MultipleContentTypeField(
+        feature="webhooks", choices_as_strings=False, required=False, label="Content Type(s)"
+    )

     class Meta:
         model = Webhook
@@ -741,7 +743,7 @@ class StatusForm(BootstrapMixin, CustomFieldModelForm, RelationshipModelForm):
     """Generic create/update form for `Status` objects."""

     slug = SlugField()
-    content_types = MultipleContentTypeField(feature="statuses", label="Content type(s)")
+    content_types = MultipleContentTypeField(feature="statuses", choices_as_strings=False, label="Content type(s)")

     class Meta:
         model = Status
@@ -785,7 +787,9 @@ class StatusBulkEditForm(BootstrapMixin, CustomFieldBulkEditForm):

     pk = forms.ModelMultipleChoiceField(queryset=Status.objects.all(), widget=forms.MultipleHiddenInput)
     color = forms.CharField(max_length=6, required=False, widget=ColorSelect())
-    content_types = MultipleContentTypeField(feature="statuses", required=False, label="Content type(s)")
+    content_types = MultipleContentTypeField(
+        feature="statuses", choices_as_strings=False, required=False, label="Content type(s)"
+    )

     class Meta:
         nullable_fields = []
(END)
```

**Comment by glennmatthews** [comment]
Thank you Jathan for catching this and for the analysis! A casualty of making "straightforward" changes after I'd already done the bulk of my manual testing. ☠️ I've reworked your diffs a little bit (I wanted `choices_as_strings` to default to `False` rather than `True`, for more clarity) and things look good now after some quick manual retesting.

**Comment by jathanism** [comment]
> Thank you Jathan for catching this and for the analysis! A casualty of making "straightforward" changes after I'd already done the bulk of my manual testing. ☠️ I've reworked your diffs a little bit (I wanted `choices_as_strings` to default to `False` rather than `True`, for more clarity) and things look good now after some quick manual retesting.

Yeah in actuality you're reminding me that we should have dedicated unit tests for each of the custom field implementations. I'm not saying we should do this now, but this turned out to be yet another case where the unit tests are not exercising how the fields would be used in practice, which is problematic.

File: nautobot/extras/forms.py [hunk 3]
```diff
@@ -769,10 +751,7 @@ class StatusForm(BootstrapMixin, CustomFieldModelForm, RelationshipModelForm):
     """Generic create/update form for `Status` objects."""
 
     slug = SlugField()
-    content_types = forms.ModelMultipleChoiceField(
-        queryset=ContentType.objects.filter(FeatureQuery("statuses").get_query()).order_by("app_label", "model"),
-        label="Content type(s)",
-    )
+    content_types = MultipleContentTypeField(feature="statuses", label="Content type(s)")
```
Comment (jathanism) [inline nautobot/extras/forms.py:754]:
```suggestion
    content_types = MultipleContentTypeField(feature="statuses", label="Content Type(s)")
```

File: nautobot/extras/forms.py [hunk 4]
```diff
@@ -808,10 +788,7 @@ class StatusFilterForm(BootstrapMixin, CustomFieldFilterForm):
     model = Status
     q = forms.CharField(required=False, label="Search")
     content_types = MultipleContentTypeField(
-        queryset=ContentType.objects.filter(FeatureQuery("statuses").get_query()).order_by("app_label", "model"),
-        required=False,
-        label="Content type(s)",
-        widget=StaticSelect2Multiple(),
+        feature="statuses", choices_as_strings=True, required=False, label="Content type(s)"
```
Comment (jathanism) [inline nautobot/extras/forms.py:791]:
```suggestion
        feature="statuses", choices_as_strings=True, required=False, label="Content Type(s)"
```

File: nautobot/extras/forms.py [hunk 5]
```diff
@@ -821,11 +798,7 @@ class StatusBulkEditForm(BootstrapMixin, CustomFieldBulkEditForm):
 
     pk = forms.ModelMultipleChoiceField(queryset=Status.objects.all(), widget=forms.MultipleHiddenInput)
     color = forms.CharField(max_length=6, required=False, widget=ColorSelect())
-    content_types = forms.ModelMultipleChoiceField(
-        queryset=ContentType.objects.filter(FeatureQuery("statuses").get_query()).order_by("app_label", "model"),
-        label="Content type(s)",
-        required=False,
-    )
+    content_types = MultipleContentTypeField(feature="statuses", required=False, label="Content type(s)")
```
Comment (jathanism) [inline nautobot/extras/forms.py:801]:
```suggestion
    content_types = MultipleContentTypeField(feature="statuses", required=False, label="Content Type(s)")
```

**Review by jathanism — APPROVED** [review]
I didn't catch this before, but while we're in here let's change the few labels that had different casing. Good to ship! :)

**Pull request merged**